### ISSUE TITLE
How to clear selected date time in react date time range picker?

### ISSUE THREAD

**Issue by oshiniG** [snamoah/react-datetime-range-picker#10]
Once user pick some date time using range picker how can they clear picked range. Once user highlight input value (picked date time)  ability to press delete button or there is close icon to clear picked date time. **input = {true}** means can enter manually. but i try to enter values manually it creates errors.

import React, { Component } from "react";
import DatetimeRangePicker from 'react-datetime-range-picker';

class Test extends Component {
   <DatetimeRangePicker
          startDate= {this.state.startDate}
          endDate={this.state.endDate}
          onStartDateBlur={this.startDateBlurHandler}
          onEndDateBlur={this.endDateBlurHandler}
          input = {true}
         timeFormat="HH:mm"
         dateFormat="DD/MM/YY"
         inputProps={{ placeholder: ('Enter Date')}} />
   }

**Comment by snamoah** [comment]
@oshiniG do you intend to select another date after clearing, you can do this by managing the states via states...
E.g.

```
<DatetimeRangePicker
        className="row"
        dateFormat="MMM D -"
        startDate={this.state.startDate.toDate()}
        endDate={this.state.endDate.toDate()}
        pickerClassName="col-xs-6"
        isValidStartDate={d => d.isSameOrAfter(new Date(), 'day')}
       onChange={this.onChangeHandler.bind(this)}
 />
```

there you're setting the initial values for startDate and endDates from a state... and the onChange handler will update these states so tohey're in turn updated in the component. do this answer your quetion?

**Comment by oshiniG** [comment]
> @oshiniG do you intend to select another date after clearing, you can do this by managing the states via states...
> E.g.
> 
> ```js
> <DatetimeRangePicker
>         className="row"
>         dateFormat="MMM D -"
>         startDate={this.state.startDate.toDate()}
>         endDate={this.state.endDate.toDate()}
>         pickerClassName="col-xs-6"
>         isValidStartDate={d => d.isSameOrAfter(new Date(), 'day')}
>        onChange={this.onChangeHandler.bind(this)}
>  />
> ```
> 
> there you're setting the initial values for startDate and endDates from a state... but there's no `onChange` that prop will update these states so they're in turn updated in the component. does this answer your question?

**Comment by oshiniG** [comment]
Mistakenly status update as closed.
@snamoah thanks for your explanation.
state update using `startDateBlurHandler` function not using `onchange` . but i can't clear picked date in every  `onchange`. there should be a specific event for that. ex: there should be a close icon or user can highlight and can press delete or backspace button from the keyboard.

- And also start date is must .if user not select start date user can't pick end date.

 following link can use for more details.
[codesandbox link]( codesandbox.io/s/7ojk94n9kx)

**Comment by oshiniG** [comment]
Happy to say it's resolved using  react-datetime-range-picker 2.0.1 version
